src/docs/sphinx/dev_guide/updating_tpls.rst

```diff
@@ -0,0 +1,188 @@
+             there is a ``compilers.yaml`` file containing compiler and 
+             version information for compilers we use for development and 
+             testing. If you wish to test and build with a new compiler or 
+             version on a system, modify the appropriate ``compilers.yaml`` 
+             file.
+
+   To change a version of a non-system TPL, go into the 
+   ``axom/scripts/spack/packages`` directory. There you will find a 
+   sub-directory for each TPL Axom uses. Modify the contents of the Spack
+   package file ``package.py`` in each package sub-directory as needed. 
+
+#. **Test installation.**
+   Do a test installation of your changes in a local directory by running
+   the following command in the top-level Axom directory::
+
+   $ ./scripts/llnl_scripts/build_tpls.py -d local/install/path
+
+   where ``local/install/path`` is a directory location where you want the 
+   libraries to be installed.
+
+   The script will do a full installation of all TPLs for the plaform you 
+   are running on. It will output whether the TPL install succeeded and, 
+   subsequently, whether an Axom build against the TPL install succeeded.
+
+   Running the script produces new host-config files (i.e., CMake cache files) 
+   that you can use to build and test Axom with the installation, if issues
+   arise. The generated host-config files will be located in the top-level Axom
+   directory of your local clone of the repo. If any changes to Axom code are 
+   needed to work with the TPL update(s), make the changes and test them.
+
+   .. note:: You can build a subset of TPLs for a platform, by passing Spack
+             spec arguments to the Python script. For example,
+
+             ``$ ./scripts/llnl_scripts/build_tpls.py --spec clang@10.0.0~cpp14+devtools+mfem+c2c ...``
+
+             will build the TPLs for the clang 10.0.0 compiler. Please see the
+             ``scripts/spack/specs.json`` file for available specs. 
+
+#. When you are confident that everything is correct, log in as user
+   ``atk`` to each of the machines named in Axom's standard host-configs and run
+
+   $ scripts/llnl/build_tpl.py
+
+   This will do all of the standard installations in the shared directories
+   used by Axom developers. When completed, they will produce new host-config
+   files for each configuration. Give these files to your regular user account
+   and log back in to that account. Copy these new host-config files to the
+   ``host-configs`` subdirectory and commit them to your branch. Make sure all
+   file changes from all previous steps are also committed and pushed upstream.
+
+#. Next, build the docker images for continuous integration using GitHub
+   actions. From Axom's GitHub page, click on "Actions" and then on "Docker
+   TPL build" in the "Workflows" menu. Find the "Run Workflow" drop-down
+   menu, select your branch, and click on the "Run workflow" button. This
+   will launch the build of the docker images.
```

Comment:
Suggestion: Can we please add a link to the github actions page?
E.g.
https://github.com/LLNL/axom/actions/workflows/docker_build_tpls.yml